flex-squid/src/main/java/org/sonar/flex/FlexVisitorContext.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SonarQube Flex Plugin
+ * Copyright (C) 2010-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.flex;
+
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.RecognitionException;
+import java.io.File;
+import javax.annotation.CheckForNull;
+
+public class FlexVisitorContext {
+
+  private final AstNode rootTree;
+  private final RecognitionException parsingException;
+
+  public FlexVisitorContext(File file, AstNode tree) {
+    this(file, tree, null);
+  }
+
+  public FlexVisitorContext(File file, RecognitionException parsingException) {
+    this(file, null, parsingException);
+  }
+
+  private FlexVisitorContext(File file, AstNode rootTree, RecognitionException parsingException) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "file". ([squid:S1172](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-105&issues=AWF7WofFGUleEzPZFxtT&open=AWF7WofFGUleEzPZFxtT)

[ISSUE_KEY]: AWF7WofFGUleEzPZFxtT